bankaccount.go

```diff
@@ -3,43 +3,106 @@ package stripe
 import (
 	"encoding/json"
 	"fmt"
+	"strconv"
+
+	"github.com/stripe/stripe-go/form"
 )
 
 // BankAccountStatus is the list of allowed values for the bank account's status.
 // Allowed values are "new", "verified", "validated", "errored".
 type BankAccountStatus string
 
-// BankAccountParams is the set of parameters that can be used when creating or updating a bank account.
+// BankAccountParams is the set of parameters that can be used when updating a
+// bank account.
+//
+// Note that while form annotations are used for updates, bank accounts have
+// some unusual logic on creates that necessitates manual handling of all
+// parameters. See AppendToAsSourceOrExternalAccount.
 type BankAccountParams struct {
-	Params
+	Params `form:"*"`
 
 	// The identifier of the parent account under which bank accounts are
 	// nested.
-	AccountID string
+	AccountID string `form:"-"`
+
+	Customer string `form:"-"`
 
 	// A token referencing an external account like one returned from
 	// Stripe.js.
-	Token string
+	Token string `form:"-"`
 
 	// Information on an external account to reference. Only used if `Token`
 	// is not provided.
-	Account, AccountHolderName, AccountHolderType, Country, Currency, Routing string
+	Account           string `form:"account_number"`
```

Comment:
Yeah, I've tried to `-` out any fields that never need to be form-encoded automatically.

> I know it might be used for updates so I wonder if it'd be cleaner to split the class in two, especially as some params only exist on create.

Hm, yeah I think that would definitely be cleaner. Can we move these changes to a separate list or something though? I've kind of made this entire change API compatible, and given the already-considerable size, it'd be nice to start moving some of those changes into separate patches.

Reply:
Totally, was hoping you'd keep track of those while working on the PR since there are likely multiple things we want to fix in the future. Out of scope for this PR I think.